### PR TITLE
Jackson serializer typed.ActorRef, #24155

### PR DESCRIPTION
**Pull request by patriknw**
* most convenient for users to include it akka-serialization-jackson
  and load it when akka-actor-typed is in classpath
* provided dependency to akka-actor-typed

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #24155

**Comment by patriknw** [comment]
cc @ignasi35

**Comment by ignasi35** [comment]
> cc @ignasi35

thanks for the ping!

I started looking at https://github.com/lagom/lagom/pull/1955 this week but got pulled into pushing the support for 2.13 for Lagom. Next couple of weeks I'll be conferencing and team meeting so I'll have little time to review this. :-(

**Review by chbatey — APPROVED** [review]
LGTM with one question of optional vs provided

File: build.sbt [hunk 1]
```diff
@@ -239,7 +239,7 @@ lazy val docs = akkaModule("akka-docs")
   .disablePlugins(ScalafixPlugin)
 
 lazy val jackson = akkaModule("akka-serialization-jackson")
-  .dependsOn(actor, actorTests % "test->test", testkit % "test->test")
+  .dependsOn(actor, actorTyped % "provided->compile", actorTests % "test->test", testkit % "test->test")
```
[the remark this answers is not in the document]
Comment (patriknw) [reply build.sbt]: agreed, now it's:
```
        <dependency>
            <groupId>com.typesafe.akka</groupId>
            <artifactId>akka-actor-typed_2.12</artifactId>
            <version>2.6-SNAPSHOT</version>
            <optional>true</optional>
        </dependency>
```

**Review by ignasi35 — APPROVED** [review]

**Pull request merged**